nasa-gibs/worldview
# GitHub Actions CI/CD

## Description

Fixes #2689

Moves all our CI/CD from Travis to GitHub actions. 

Since `virtualenv` was giving me trouble on the Windows Github Actions runners, I tried out `pipenv` to handle both managing Python dependencies and the project-level virtual environment.  Since this seems to work well and consistently across all platforms, I replaced our usage of `virtualenv` and manual `PATH` setting with `pipenv`. Python dependencies are now defined in the `Pipfile` and `Pipfile.lock` which work almost identically to the NPM `package.json` and `package-lock.json`.

Running some Python code within the virtual environment is accomplished by prefixing several node scripts in the `package.json` with `pipenv run`.

The `.python` directory can be removed entirely since that was being used by `virtualenv`.

## How To Test

1. Remove your `.python` directory in the project root, it is no longer needed
2. Run `npm run python` to install `pipenv` and have `pipenv` install the project dependencies and virtual environment
3. Run `npm run build` and confirm project builds properly.
